### PR TITLE
feat: Implementation of event invoice generation & calculation

### LINKED ISSUE #6120: Implement module to show paid, due and upcoming invoices
**Is your feature request related to a problem? Please describe.**

This is a sub issue of #6101 .
 For the implementation of paying invoices, the organizer needs to be able to go through their paid, due and upcoming invoices. For this, currently there is no internal implementation. Helper functions are required.
Through this server side implementation, the FE integration can be expedited.

**Additional context**
Working on it.

### LINKED ISSUE #6124: Implement function for event invoice calculation & invoice generation
**Describe the feature**
To calculate event invoice and send it to the organizer, the event invoice needs to be calculated and scheduled for every 1st of the month. This can then be rendered onto an event invoice PDF and become available for the organizer.
A job can scheduled on the 1st of every month. The invoice generation function will be associated
with the job.
**Additional context**
Working on it.

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #6120
Fixes #6124 

#### Short description of what this resolves:

This implements a module for event invoice where the monthly gross amount & net amount are calculated and sent to every organizer on the 1st of every month.
#### Changes proposed in this pull request:

- Provides helper functions to show invoices on organizer dashboard
- Calculates event invoice
- Schedules event invoice generation job for every 1st of the month.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by mrsaicharan1** [comment]
@uds5501 @shreyanshdwivedi Please have a look

**Comment by iamareebjamal** [comment]
Implement completely, you have just added helpers

**Comment by mrsaicharan1** [comment]
I was waiting for the FE integration to be done so that I could test it
out. Is that okay?

**Comment by mrsaicharan1** [comment]
@iamareebjamal Is it okay if I implement the invoice generation in a follow up PR?

**Comment by iamareebjamal** [comment]
I don't recommend breaking up same feature into multiple PRs. 1 commit = 1 complete feature

**Comment by mrsaicharan1** [comment]
Alright will do it this itself

**Comment by iamareebjamal** [comment]
Solve codacy issues

File: app/__init__.py [hunk 1]
```diff
@@ -242,6 +242,7 @@ def update_sent_state(sender=None, headers=None, **kwargs):
 scheduler.add_job(send_event_fee_notification_followup, 'cron', day=15)
 scheduler.add_job(change_session_state_on_event_completion, 'cron', hour=5, minute=30)
 scheduler.add_job(expire_pending_tickets, 'cron', minute=45)
+scheduler.add_job(send_monthly_event_invoice, 'cron', day=1, month=1-12)
```
Comment (mrsaicharan1) [inline app/__init__.py]: Scheduled this job for every first day of each month.

**Comment by mrsaicharan1** [comment]
@iamareebjamal @shreyanshdwivedi @uds5501 Please have a look. If this fine, will continue to implement further sub-issues linked to this.

**Comment by mrsaicharan1** [comment]
I think I need modify the query such that the ticket sales which are queried should Only be restricted to that particular month, for a monthly cycle

**Comment by mrsaicharan1** [comment]
@iamareebjamal @shreyanshdwivedi @uds5501 Please review.

**Comment by mrsaicharan1** [comment]
@iamareebjamal @shreyanshdwivedi @uds5501

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 @iamareebjamal @uds5501 @shreyanshdwivedi Please have a look as this is a requirement for # to move forward.

**Comment by iamareebjamal** [comment]
Fixes #abc, #xyz 

doesnt work in GitHub. Separate issues in separate lines

**Pull request merged**